cmd/buildctl/common/common.go

```
@@ -18,6 +18,28 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
+// ResolveTLSFiles scans a TLS directory for known cert/key filenames.
+func ResolveTLSFilesFromDir(tlsDir string) (caCert, cert, key string) {
+	// Look for ca.pem or ca.crt and, if it exists, set caCert to that
+	// Look for cert.pem or tls.crt and, if it exists, set cert to that
+	// Look for key.pem or tls.key and, if it exists, set key to that
+	for _, v := range [6]string{"ca.pem", "cert.pem", "key.pem", "ca.crt", "tls.crt", "tls.key"} {
+		file := filepath.Join(tlsDir, v)
+		if _, err := os.Stat(file); err == nil {
+			switch v {
+			case "ca.pem", "ca.crt":
+				caCert = file
+			case "cert.pem", "tls.crt":
+				cert = file
+			case "key.pem", "tls.key":
+				key = file
+			}
+		}
```

Can you readd the `else` and do it like this?

```go
if _, err := os.Stat(file); err == nil {
    // original code
} else if !errors.Is(err, os.ErrNotExist) {
   return nil, err
}
```

This will restrict it so when the error is something like permission denied it will still error so we don't end up just ignoring any error. If the error is that the file couldn't be found, we'll just ignore it.

---

Only checking `ErrNotExist` may cause issue as sometimes invalid paths will return EPERM (or EISDIR).

I think we need to probably just read it in groups of 3 and if both groups fail then show error.

---

I changed the logic to read by groups of 3, while doings so I chose a naive approach of "Stat fails == bad", without caring too much about the exact error.
I wonder if it's fine, or maybe it would be better to add a hint to the error message that there might be a permissions error involved?